src/components/MarqueeSelection/MarqueeSelection.Props.ts

```diff
@@ -0,0 +1,28 @@
+import * as React from 'react';
+import { ISelection } from '../../utilities/selection/interfaces';
+import { MarqueeSelection } from './MarqueeSelection';
+
+export interface IMarqueeSelectionProps extends React.Props<MarqueeSelection> {
+  /**
+   * The selection object to interact with when updating selection changes.
+   */
+  selection: ISelection;
+
+  /**
+   * The base element tag name to render the marquee bounding area within.
+   * @default div
+   */
+  rootTagName?: string;
+
+  /**
+   * Optional props to mix into the root element.
+   */
+  rootProps?: React.HTMLProps<HTMLDivElement>;
+
+  /**
+   * Optional callback that is called, when the mouse down event occurs, in order to determine
+   * if we should start a marquee selection. If true is returned, we will cancel the mousedown
+   * event to prevent upstream mousedown handlers from executing.
+   */
+  onShouldStartSelection?: (ev: React.MouseEvent) => boolean;
```

Comment:
I'm not sure that the rectangle helps at all. It would force the caller to figure out whats in the box, which pushes some of the logic in more than one place. However, it is possible we could try to identify if the pixel you clicked on is an item and pass that as a parameter. That could make it cleaner.